### DISCUSSION TITLE
Debugger not stopping on breakpoints for Kafka listeners

### DISCUSSION THREAD

**Discussion by cbfaucherupgrade** [metalbear-co/mirrord#1935] [Q&A] [answered]
Greetings, 
I've been experimenting with MirrorD, and so far I love it.

Once connected, breakpoints work fine for ReST and GRPC endpoints.

However, breakpoints fail to stop when on Kafka listeners.  More specifically, on methods annotated with Spring `@KafkaListener` .

Is this a known issue?  Any workarounds?

**Comment by aviramha** [comment, marked as answer]
Hi @cbfaucherupgrade - Happy to hear that you love mirrord!
Do you see your local service getting the messages? It is very likely that since Kafka is a service you connect to and retrieve messages from, your local service is added as another replica/consumer in the group, and if there isn't an available partition to read from it'd be idle, since Kafka can have max consumers on consumer group equal to the partitions count.
If you want to read the messages instead of the deployed service, you can use the pause feature to make the targeted pod frozen while you use mirrord, then it'll replace it as a consumer and so get messages.
More info - https://mirrord.dev/docs/guides/pause/

**Comment by cbfaucherupgrade** [comment]
I see.   I thought the mirroring of inputs/outputs would cover for that.
I'm testing with a topic having a single partition.  So that partition is assigned to the cluster instance, not the local one.
Will check this "pause" option, and report back.

**Reply by cbfaucherupgrade** [reply]
...And `--pause` seems to do the trick! 🎉 
Thanks for the quick help!
